### PR TITLE
release: Nvim 0.5.1

### PR DESCRIPTION
**Pull request by justinmk**
Please test this locally:

```
VIMRUNTIME=/path/to/neovim/runtime /path/to/neovim/build/bin/nvim
```

## Notes

- 14cdaca6a8180c6e3079cd2e73fa75a5eb5253da fixes a problem not noticed on master because of the vim.lsp.diagnostic => vim.diagnostic move
    - also fixes some tests that accidentally depended on the previous session

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -137,8 +137,8 @@ set_property(CACHE CMAKE_BUILD_TYPE PROPERTY
 # version string, else they are combined with the result of `git describe`.
 set(NVIM_VERSION_MAJOR 0)
 set(NVIM_VERSION_MINOR 5)
-set(NVIM_VERSION_PATCH 0)
-set(NVIM_VERSION_PRERELEASE "") # for package maintainers
+set(NVIM_VERSION_PATCH 2)
```
Comment (mjlbach) [inline CMakeLists.txt:140]: Minor question, why is this 2?

Comment (clason) [reply CMakeLists.txt:140]: I think this is for _after_ the release? (So a build from `release-0.5` reports itself as `0.5.2-dev` -- see the following line.)

Comment (justinmk) [reply CMakeLists.txt:140]: it should have been 1 before.

Comment (mjlbach) [reply CMakeLists.txt:140]: Got it, thanks!

**Review by jamessan — APPROVED** [review]

**Comment by justinmk** [comment]
`CI / linux functionaltest-lua (cc=gcc)` for reference:

```
021-09-26T16:05:44.4420590Z [32m[2m[----------][0m[0m Running tests from [36mtest/functional/plugin/lsp_spec.lua[0m
...
2021-09-26T16:05:44.8291539Z [32m[2m[ RUN      ][0m[0m LSP basic_init test should succeed with manual shutdown: [2m50.58 ms[0m [1m[32mOK[0m[0m
2021-09-26T16:23:20.4923045Z -- Tests exited non-zero: Process terminated due to timeout
2021-09-26T16:23:20.4923982Z CMake Error at /home/runner/work/neovim/neovim/cmake/RunTests.cmake:90 (message):
2021-09-26T16:23:20.4924843Z   functional tests failed with error: Process terminated due to timeout
```

File: CMakeLists.txt (hunk 1 above)
Comment (bfredl) [reply CMakeLists.txt:140]: because I dun goofed

**Comment by justinmk** [comment]
this test is timing out:

> LSP basic_init test should check the body sent attaching before init

**Pull request merged**